dspace-services/src/main/resources/spring/spring-dspace-core-services.xml

```diff
@@ -16,18 +16,33 @@
     <!-- Place all DSpace core service bean definitions below here -->
 
     <!-- PROPERTY PLACEHOLDER SETUP -->
-    <!-- The following two beans are a cool trick. They let us use dynamic properties
-         like ${dspace.dir} within our Spring bean XML configuration files. These
-         property values are then dynamically loaded via our DSpaceConfigurationService. -->
-    <!-- First, configure a property placeholder. This lets us use properties like ${..} within Spring bean configs -->
-    <bean id="propertyPlaceholderConfigurer"
-          class="org.springframework.beans.factory.config.PropertyPlaceholderConfigurer">
-        <property name="properties" ref="dspaceConfigurationFactoryBean"/>
+    <!-- Configure the Commmons Configuration "ConfigurationPropertiesFactoryBean" to use our DSpace Configuration
+         service to perform Property Placeholder substitution in Spring configuration. For example, this lets us
+         use dynamic properties like ${dspace.dir} within our Spring bean XML configuration files. These property
+         values are then dynamically loaded via our DSpaceConfigurationService at runtime. -->
+    <!-- See: https://commons.apache.org/proper/commons-configuration/userguide/howto_utilities.html#Use_Configuration_in_Spring -->
+    <bean id="propertyPlaceholderConfigurer" class="org.springframework.beans.factory.config.PropertyPlaceholderConfigurer">
+        <property name="properties">
+            <bean class="org.apache.commons.configuration2.spring.ConfigurationPropertiesFactoryBean">
+                <constructor-arg>
+                    <ref bean="dspaceConfiguration" />
+                </constructor-arg>
+            </bean>
+        </property>
+    </bean>
```

Comment:
Do we need this *and* PropertySourcesPlaceholderConfigurer (which is meant to replace it)?

Reply:
Good question. I actually overlooked the [PropertyPlaceholderConfigurer](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/PropertyPlaceholderConfigurer.html) now has a reference to "use [PropertySourcesPlaceholderConfigurer](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/support/PropertySourcesPlaceholderConfigurer.html)" as a replacement.

You are correct that I've configured *both* here.  I'll try out removing the former, and see if all our tests still pass (which should tell us if we actually need both here or not).  As you likely saw, I have separate unit tests that already test for the features provided by these `Configurer` classes are "working" properly with our ConfigurationService.

Reply:
Good eye here, We can remove the PropertyPlaceholderConfigurer entirely, and *just use* the PropertySourcesPlaceholderConfigurer.  I've done that in the latest commit & all tests pass.